test/support/fake_erlang.ex

```diff
@@ -0,0 +1,5 @@
+defmodule FakeErlang do
+  use TestAgent, %{system_info: 'x86_64-apple-darwin20.2.0'}
+
+  def system_info(_), do: get(__MODULE__, :system_info)
```

Comment:
It might be better for the method implementation to respond only to `system_info(:system_architecture)`. Turns out [you can ask `:erlang.system_info/1` about lots of things](https://www.erlang.org/doc/man/erlang.html#system_info-1), and I think getting an error if that happens would be clearer.

Reply:
I created a PR for this: https://github.com/appsignal/appsignal-elixir/pull/760